src/schema/schema-generator.ts

```diff
@@ -663,12 +670,26 @@ export abstract class SchemaGenerator {
   private static getResolveTypeFunction<TSource = any, TContext = any>(
     resolveType: TypeResolver<TSource, TContext>,
   ): GraphQLTypeResolver<TSource, TContext> {
-    return async (...args) => {
-      const resolvedType = await resolveType(...args);
-      if (typeof resolvedType === "string") {
-        return resolvedType;
+    return (...args) => {
+      const resolvedTypeOrPromise = resolveType(...args);
+
+      if (isPromiseLike(resolveType)) {
```

Comment:
I think it should be `resolvedTypeOrPromise` - we are checking the return value of `resolveType` fn